handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -235,37 +235,39 @@ protected void deallocate() {
         } finally {
             readerLock.unlock();
         }
-        ssl = SSL.newSSL(context.ctx, !context.isClient());
-        try {
-            networkBIO = SSL.bioNewByteBuffer(ssl, context.getBioNonApplicationBufferSize());
+        synchronized (this) {
+            ssl = SSL.newSSL(context.ctx, !context.isClient());
```

Comment:
Done.  `this.ssl` has to be assigned fairly early on, diminishing the safety of accidentally referring to it due to other methods in the constructor referencing it (yikes!).